apps/sozip.cpp

```diff
@@ -367,173 +341,142 @@ static int Validate(const char *pszZipFilename, bool bVerbose)
-            aosOptions.SetNameValue("SOZIP_ENABLED", papszArgv[iArg]);
-        }
-        else if (STARTS_WITH(papszArgv[iArg], "--enable-sozip="))
-        {
-            aosOptions.SetNameValue(
-                "SOZIP_ENABLED", papszArgv[iArg] + strlen("--enable-sozip="));
-        }
-        else if (strcmp(papszArgv[iArg], "--sozip-chunk-size") == 0 &&
-                 iArg + 1 < nArgc)
-        {
-            ++iArg;
-            aosOptions.SetNameValue("SOZIP_CHUNK_SIZE", papszArgv[iArg]);
-        }
-        else if (STARTS_WITH(papszArgv[iArg], "--sozip-chunk-size="))
-        {
-            aosOptions.SetNameValue("SOZIP_CHUNK_SIZE",
-                                    papszArgv[iArg] +
-                                        strlen("--sozip-chunk-size="));
-        }
-        else if (strcmp(papszArgv[iArg], "--sozip-min-file-size") == 0 &&
-                 iArg + 1 < nArgc)
-        {
-            ++iArg;
-            aosOptions.SetNameValue("SOZIP_MIN_FILE_SIZE", papszArgv[iArg]);
-        }
-        else if (STARTS_WITH(papszArgv[iArg], "--sozip-min-file-size="))
-        {
-            aosOptions.SetNameValue("SOZIP_MIN_FILE_SIZE",
-                                    papszArgv[iArg] +
-                                        strlen("--sozip-min-file-size="));
-        }
-        else if (strcmp(papszArgv[iArg], "--content-type") == 0 &&
-                 iArg + 1 < nArgc)
-        {
-            ++iArg;
-            aosOptions.SetNameValue("CONTENT_TYPE", papszArgv[iArg]);
-        }
-        else if (STARTS_WITH(papszArgv[iArg], "--content-type="))
-        {
-            aosOptions.SetNameValue(
-                "CONTENT_TYPE", papszArgv[iArg] + strlen("--content-type="));
-        }
-        else if (papszArgv[iArg][0] == '-')
-        {
-            Usage(true, CPLSPrintf("Unhandled option %s", papszArgv[iArg]));
-        }
-        else if (pszZipFilename == nullptr)
-        {
-            pszZipFilename = papszArgv[iArg];
-        }
-        else
-        {
-            aosFiles.AddString(papszArgv[iArg]);
-        }
+        auto &group = argParser.add_mutually_exclusive_group();
+        group.add_argument("-g", "--grow")
+            .flag()  // Default mode. Nothing to do
+            .help(
+                _("Grow an existing zip file with the content of the specified "
+                  "filename(s). Default mode"));
+        group.add_argument("--overwrite")
+            .store_into(bOverwrite)
+            .help(_("Overwrite the target zip file if it already exists."));
     }
 
-    if (!pszZipFilename)
+    bool bList = false;
+    bool bValidate = false;
+    std::string osOptimizeFrom;
+    std::vector<std::string> aosFiles;
     {
-        Usage(true, "Missing zip filename");
-        return 1;
+        auto &group = argParser.add_mutually_exclusive_group();
+        group.add_argument("-l", "--list")
+            .store_into(bList)
+            .help(_("List the files contained in the zip file"));
```

Comment:
also check other strings for consistency
```suggestion
            .help(_("List the files contained in the zip file."));
```

Reply:
done